Automattic/wp-calypso
# Style selection: Update style variation UI

#### Proposed Changes

This PR updates the style preview UI in the design preview component. Previously, we couldn't update the style preview UI without updating the package `@wordpress/edit-site`, which is a complicated ordeal (see p1666235331775949-slack-C03DY3RJ5 and pbxlJb-2gk-p2#comment-1952). Also, see the abandoned PR that attempts to do so: #69272.

As an alternative solution, instead of updating `@wordpress/edit-site`, we make a local copy of the [component](https://github.com/WordPress/gutenberg/blob/7005d47340dd9fc6657f2a1c188c6dbf3f3d6dfb/packages/edit-site/src/components/global-styles/preview.js) `<StylesPreview />` instead of directly importing it from `@wordpress/edit-site`. The downside of this approach is that we no longer benefit from the abstraction provided by the package, by the upside is that we can now update the style preview UI without needing to bump `@wordpress/edit-site` and its dependencies across Calypso. This approach can also potentially prevent us from having to update `@wordpress/edit-site` each time the style preview UI changes.

#### Testing Instructions



* Head to the design picker `setup/designSetup?siteSlug=${site_slug}`
* Open any design with style variation.
* Ensure that the style preview UI is updated to match the one in the site editor (wordpress.com/site-editor/${site_slug})

#### Pre-merge Checklist
